Dockerfile

```diff
@@ -5,9 +5,26 @@
 # docs/source/dev/dockerfile/dockerfile.rst and
 # docs/source/assets/dev/dockerfile-stages-dependency.png
 
+ARG CUDA_VERSION=12.4.1
 #################### BASE BUILD IMAGE ####################
 # prepare basic build environment
-FROM nvidia/cuda:12.4.1-devel-ubuntu22.04 AS dev
+FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu22.04 AS base
+
+ARG CUDA_VERSION
+ENV CUDA_VERSION=${CUDA_VERSION}
```

Comment:
Is this repeated? Declared in line 8

Reply:
The one on line 8 is to declare `CUDA_VERSION` so `FROM nvidia/cuda:${CUDA_VERSION}` can reference it. This is defined within the build stage so other steps can reference to `CUDA_VERSION`